exercises/robot-simulator/src/test/java/RobotTest.java

```diff
@@ -0,0 +1,211 @@
+        final Robot robot = new Robot(new GridPosition(0, 0), Orientation.SOUTH);
+
+        robot.advance();
+
+        final GridPosition expectedGridPosition = new GridPosition(0, -1);
+        final GridPosition actualGridPosition = robot.getGridPosition();
+
+        assertTrue(actualGridPosition.equals(expectedGridPosition));
+    }
+
+    @Ignore
+    @Test
+    public void testAdvancingWhenFacingEastIncreasesXCoordinateByOne() {
+        final Robot robot = new Robot(new GridPosition(0, 0), Orientation.EAST);
+
+        robot.advance();
+
+        final GridPosition expectedGridPosition = new GridPosition(1, 0);
+        final GridPosition actualGridPosition = robot.getGridPosition();
+
+        assertTrue(actualGridPosition.equals(expectedGridPosition));
+    }
+
+    @Ignore
+    @Test
+    public void testAdvancingWhenFacingWestDecreasesXCoordinateByOne() {
+        final Robot robot = new Robot(new GridPosition(0, 0), Orientation.WEST);
+
+        robot.advance();
+
+        final GridPosition expectedGridPosition = new GridPosition(-1, 0);
+        final GridPosition actualGridPosition = robot.getGridPosition();
+
+        assertTrue(actualGridPosition.equals(expectedGridPosition));
+    }
+
+    @Ignore
+    @Test
+    public void testInstructionsToMoveWestAndNorth() {
+        testInstructions(
+                new GridPosition(0, 0),
+                Orientation.NORTH,
+                "LAAARALA",
+                new GridPosition(-4, 1),
+                Orientation.WEST);
+    }
+
+    @Ignore
+    @Test
+    public void testInstructionsToMoveWestAndSouth() {
+        testInstructions(
+                new GridPosition(2, -7),
+                Orientation.EAST,
+                "RRAAAAALA",
+                new GridPosition(-3, -8),
+                Orientation.SOUTH);
+    }
+
+    @Ignore
+    @Test
+    public void testInstructionsToMoveEastAndNorth() {
+        testInstructions(
+                new GridPosition(8, 4),
+                Orientation.SOUTH,
+                "LAAARRRALLLL",
+                new GridPosition(11, 5),
+                Orientation.NORTH);
+    }
+
+    // Helper Methods
+
+    private void testTurningRight(final Orientation startOrientation, final Orientation expectedEndOrientation) {
+        final Robot robot = new Robot(new GridPosition(0, 0), startOrientation);
+
+        robot.turnRight();
+
+        final Orientation actualEndRobotOrientation = robot.getOrientation();
+
+        assertTrue(actualEndRobotOrientation.equals(expectedEndOrientation));
+    }
+
+    private void testTurningLeft(final Orientation startOrientation, final Orientation expectedEndOrientation) {
+        final Robot robot = new Robot(new GridPosition(0, 0), startOrientation);
+
+        robot.turnLeft();
+
+        final Orientation actualEndRobotOrientation = robot.getOrientation();
+
+        assertTrue(actualEndRobotOrientation.equals(expectedEndOrientation));
+    }
+
+    private void testInstructions(
```

Comment:
@stkent @jtigger in general I like the fact that these helper methods are pulled out, but it does make me squint a little when following the tests.  There's definitely a school of thought that says tests should be dead simple, even to the point of eschewing any logic to create data objects programmatically for testing.  That camp would prefer to see everything spelled out literally.

I don't have a strong feeling about this either way, but we should just consider the audience.  The tests contain a considerable amount of the guidance that the users get, we should make sure they're as useful as possible.